File: j2k/testData/fileOrElement/nullability/notNullCast.java
public class Passenger {
    public static class PassParent {
    }

    public static class PassChild extends PassParent {
    }

    public PassParent provideNullable(int p) {
        return p > 0 ? new PassChild() : null;
    }

    public void context() {
        PassParent pass = provideNullable(1);
        assert pass != null;
        accept((PassChild) pass);

        PassParent pass2 = provideNullable(1);
        if (1 == 2) {
            assert pass != null;
            accept((PassChild) pass2);
        }
        accept((PassChild) pass2);
    }

    public void accept(PassChild p) {
    }
}

File: j2k/testData/fileOrElement/nullability/notNullCast.kt
class Passenger {
    open class PassParent

    class PassChild : PassParent()

    fun provideNullable(p: Int): PassParent? {
        return if (p > 0) PassChild() else null
    }

    fun context() {
        val pass = provideNullable(1)!!
        accept(pass as PassChild?)

        val pass2 = provideNullable(1)
        if (1 == 2) {
            assert(pass != null)
            accept(pass2 as PassChild?)
        }
        accept(pass2 as PassChild?)
    }

    fun accept(p: PassChild?) {
    }
}